pkg/apis/extensions/validation/validation.go

```diff
@@ -289,7 +289,8 @@ func ValidateDeploymentSpec(spec *extensions.DeploymentSpec, fldPath *field.Path
 	}
 	if spec.ProgressDeadlineSeconds != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.ProgressDeadlineSeconds), fldPath.Child("progressDeadlineSeconds"))...)
-		if *spec.ProgressDeadlineSeconds <= spec.MinReadySeconds {
+		if *spec.ProgressDeadlineSeconds > 0 && *spec.ProgressDeadlineSeconds <= spec.MinReadySeconds {
```

Comment:
Is it safe to relax validation to allow 0? The [API guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#backward-compatibility-gotchas) say, "Validation rules on spec fields can neither be relaxed nor strengthened."

In this case, I'm not too worried about clients being broken by 0, but what happens on the server if, for example, you roll back to a previous patch release after some objects have been persisted with a 0 value? Will the object become inaccessible? Will a "list all" fail? Or does the server handle this gracefully somehow?

Reply:
Even if the object is still accessible with `0` value after downgrade, the old controller will treat deadline == 0 as "immediate pass deadline" instead of "no deadline". 

It's safer to use max int instead, then.

Reply:
Updated to use max int32 instead.